packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1100,15 +1100,39 @@ function pushStartPreformattedElement(
   responseState: ResponseState,
   assignID: null | SuspenseBoundaryID,
 ): ReactNodeList {
-  const children = pushStartGenericElement(
-    target,
-    props,
-    tag,
-    responseState,
-    assignID,
-  );
+  target.push(startChunkForTag(tag));
+
+  let children = null;
+  let innerHTML = null;
+  for (const propKey in props) {
+    if (hasOwnProperty.call(props, propKey)) {
+      const propValue = props[propKey];
+      if (propValue == null) {
+        continue;
+      }
+      switch (propKey) {
+        case 'children':
+          children = propValue;
+          break;
+        case 'dangerouslySetInnerHTML':
+          innerHTML = propValue;
+          break;
+        default:
+          pushAttribute(target, responseState, propKey, propValue);
+          break;
+      }
+    }
+  }
+  if (assignID !== null) {
+    pushID(target, responseState, assignID, props.id);
+  }
+
+  target.push(endOfStartTag);
 
-  if (typeof children === 'string' && children[0] === '\n') {
+  if (
+    (typeof children === 'string' && children[0] === '\n') ||
+    (typeof innerHTML === 'string' && innerHTML[0] === '\n')
```

Comment:
I'm confused. Isn't `innerHTML` an object here?

Reply:
Yea you're right that's a bug.

Reply:
Fixed.